### PR TITLE
 feat: Created Stripe App and implemented Stripe Webhook.

### PR DESCRIPTION
**Pull request by HammadAhmadWaqas**

### CODE REVIEW

**Review by grmartin — APPROVED** [review]
LGTM

File: commerce_coordinator/apps/stripe/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class SignatureVerificationAPIError(APIException):
+    status_code = 403
+    default_detail = 'Invalid Stripe signature'
+    default_code = 'invalid_signature'
+
+
+class InvalidPayloadAPIError(APIException):
+    status_code = 400
+    default_detail = 'Invalid stripe payload'
```
Comment (pshiu) [inline commerce_coordinator/apps/stripe/exceptions.py:14]: Nit:
```suggestion
    default_detail = 'Invalid Stripe payload'
```

File: commerce_coordinator/apps/stripe/migrations/__init__.py
Comment (pshiu) [inline commerce_coordinator/apps/stripe/migrations/__init__.py:1]: I know we have migrations/ in other apps, but unless we have any foreseeable need for the file, can we remove it to keep the directory tidy?

File: commerce_coordinator/apps/stripe/models.py
Comment (pshiu) [inline commerce_coordinator/apps/stripe/models.py:1]: I know we have models.py in other apps, but unless we have any foreseeable need for the file, can we remove it to keep the directory tidy?

File: commerce_coordinator/apps/stripe/signals.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+"""
+payment signals and receivers.
```
Comment (pshiu) [inline commerce_coordinator/apps/stripe/signals.py:2]: Nit:
```suggestion
Stripe signals and receivers.
```

File: commerce_coordinator/apps/stripe/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+"""Constants for frontend_app_payment app tests."""
```
Comment (pshiu) [inline commerce_coordinator/apps/stripe/tests/__init__.py:1]: Nit:
```suggestion
"""Constants for stripe app tests."""
```

File: commerce_coordinator/apps/stripe/views.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+endpoint_secret = settings.PAYMENT_PROCESSOR_CONFIG['edx']['stripe']['webhook_endpoint_secret']
+
+
+class WebhookView(APIView):
+    """
+    Endpoint for Stripe webhook events. A 200 response should be returned as soon as possible
+    since Stripe will retry the event if no response is received.
+
+    Django's default cross-site request forgery (CSRF) protection is disabled,
+    request are verified instead by the presence of request headers STRIPE_SIGNATURE.
```
Comment (pshiu) [inline commerce_coordinator/apps/stripe/views.py:35]: Thanks for this note!

**Review by pshiu — APPROVED** [review]
Looks great, thanks Hammad!

File: commerce_coordinator/apps/titan/tasks.py [hunk 5]
```diff
@@ -75,8 +83,21 @@ def payment_processed_save_task(payment_number, payment_state, response_code):
...
+            response_code=response_code
+        )
+        # Set cache after successfully updating payment state in Titan's system.
+        payment_state = payment['state']
+        if payment_state == PaymentState.COMPLETED.value:
+            payment_state_paid_cache_key = get_paid_payment_state_cache_key(payment_number)
+            TieredCache.set_all_tiers(payment_state_paid_cache_key, payment, settings.DEFAULT_TIMEOUT)
+        elif payment_state == PaymentState.FAILED.value:
+            payment_state_processing_cache_key = get_processing_payment_state_cache_key(payment_number)
+            TieredCache.set_all_tiers(payment_state_processing_cache_key, payment, settings.DEFAULT_TIMEOUT)
```
Comment (HammadAhmadWaqas) [inline commerce_coordinator/apps/titan/tasks.py:99]: after updating the payment in Titan, We are setting the cache.
@pshiu Thank you for reminding me.

**Pull request merged**